packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -617,6 +615,12 @@ export function getValidatorApi({
         );
       }
 
+      if (builder.status === "rejected" && engine.status === "rejected") {
+        const failureSource =
+          isEngineEnabled && isBuilderEnabled ? "Builder and engine both" : isEngineEnabled ? "Engine" : "Builder";
+        throw Error(`${failureSource} failed to produce the block`);
```

Comment:
have you tried inlining that, I feel like assigning to a variables is unnecessary unless it format in really bad, less readable if inlined. I would expect it just a oneliner